builtin/game/misc.lua

```diff
@@ -74,18 +74,37 @@ function core.after(time, func, ...)
 	}
 end
 
-function core.check_player_privs(name, privs)
+function core.check_player_privs(player_or_name, ...)
+	local name = player_or_name
+	-- Check if we have been provided with a Player object.
+	if name.get_player_name ~= nil then
```

Comment:
You should just compare the type with `"string"` or `"userdata"` (player).  The metatable lookup is unneeded, slower, and potentially error prone (consider that the `string` metatable `__index` could contain a `get_player_name` method too, although that's admitedly unlikely).